services/api-server/tests/unit/pact_broker/test_licensed_items.py

```diff
@@ -0,0 +1,233 @@
+            },
+            {
+                "category_display": "Animal",
+                "category_id": "AnimalWholeBody",
+                "source": {
+                    "available_from_url": None,
+                    "description": "Animal Models - Posable Big Male Rat V1-x",
+                    "doi": "10.13099/VIP91101-01-1",
+                    "features": {
+                        "date": "2018-01-22",
+                        "functionality": "Posable",
+                        "height": "260 mm",
+                        "name": "Big Male Rat",
+                        "sex": "male",
+                        "species": "Rat",
+                        "version": "1.0",
+                        "weight": "567 g",
+                    },
+                    "id": 111,
+                    "license_key": "MODEL_RAT567_M",
+                    "license_version": "V1.0",
+                    "protection": "Code",
+                    "thumbnail": "https://itis.swiss/assets/images/Virtual-Population/Animals-Cropped/BigMaleRat567g.png",
+                },
+                "terms_of_use_url": "https://raw.githubusercontent.com/ITISFoundation/licenses/refs/heads/main/models/User%20License%20Animal%20Models%20v1.x.md",
+            },
+        ],
+        "modified_at": "2025-02-19T13:46:30.302673Z",
+        "pricing_plan_id": 21,
+        "version": "1.0.0",
+    },
+]
+
+
+EXPECTED_LICENSED_ITEMS_PAGE = LicensedItemRpcGetPage(
+    items=[LicensedItemRpcGet.model_validate(item) for item in EXPECTED_LICENSED_ITEMS],
+    total=len(EXPECTED_LICENSED_ITEMS),
+)
+
+
+class DummyRpcClient:
+    pass
+
+
+@pytest.fixture
+async def mock_wb_api_server_rpc(app: FastAPI, mocker: MockerFixture) -> MockerFixture:
+
+    app.dependency_overrides[get_wb_api_rpc_client] = lambda: WbApiRpcClient(
+        _client=DummyRpcClient()
+    )
+
+    mocker.patch(
+        "simcore_service_api_server.services_rpc.wb_api_server._get_licensed_items",
+        return_value=EXPECTED_LICENSED_ITEMS_PAGE,
+    )
+
+    return mocker
+
+
+@pytest.fixture()
+def run_test_server(
+    get_free_port: int,
+    app: FastAPI,
+):
+    """
+    Spins up a FastAPI server in a background thread and yields a base URL.
+    The 'mocked_catalog_service' fixture ensures the function is already
+    patched by the time we start the server.
+    """
+    # Override
+    app.dependency_overrides[get_current_identity] = mock_get_current_identity
+
+    port = get_free_port
+    base_url = f"http://localhost:{port}"
+
+    config = uvicorn.Config(
+        app,
+        host="localhost",
+        port=port,
+        log_level="info",
+    )
+    server = uvicorn.Server(config)
+
+    thread = Thread(target=server.run, daemon=True)
+    thread.start()
+
+    # Wait a bit for the server to be ready
+    sleep(1)
+
+    yield base_url  # , before_server_start
+
+    server.should_exit = True
```

Comment:
but this is inconsistent with  `pact_broker_credentials`. There you are already checking for this (even a more complete check that includes CLI!).  Perhaps you want to pytest.skip in  `pact_borker_credientials` isntead of in here
Moreover, the env-vars are not the only way to get these values here